src/component/spinner.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import {
+  ActivityIndicator,
+  StyleSheet,
+  View,
+  ART,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import { color } from './style';
+
+//
+// Small Spinner
+//
+
+const smallStyles = StyleSheet.create({
+  spinner: {
+    transform: [{ scale: 1.42 }],
+  },
+});
+export const SmallSpinner = ({ ...props }) => (
+  <ActivityIndicator
+    size="small"
+    color={color.purple}
+    style={smallStyles.spinner}
+    {...props}
+  />
+);
+
+//
+// Timed Spinner
+//
+const { Surface, Shape, Path, Group } = ART;
+const width = 10;
+const linecap = 'butt';
+const spinnerColor = color.purple;
+
+export const TimedSpinner = ({ size, percent }) => (
+  <View>
+    <Surface width={size} height={size}>
+      <Group rotation={0} originX={size / 2} originY={size / 2}>
+        <Shape d={calcCirclePath(size, percent)}
+               stroke={spinnerColor}
+               strokeWidth={width}
+               strokeCap={linecap}/>
+      </Group>
+    </Surface>
+  </View>
+);
+
+TimedSpinner.propTypes = {
+  size: PropTypes.number,
+  percent: PropTypes.number.isRequired,
+};
+
+const calcCirclePath = (size, percent) => {
+  const fill = extractFill(percent);
+  return getCirclePath(
+    size / 2,
+    size / 2,
+    size / 2 - width / 2,
+    0,
+    360 * .9999 * fill / 100
+  );
+};
+
+const getCirclePath = (cx, cy, r, startDegree, endDegree) => {
+  let endDegreeInRadians = endDegree * Math.PI / 180;
+  let p = Path();
+  p.move(cx + r, cy);
+  p.arc(
+    -r + r * Math.cos(endDegreeInRadians),
+    r * Math.sin(endDegreeInRadians),
+    r,
+    r,
+    endDegree >= 180
+  );
+  return p;
+};
```

Comment:
Nice. Could we begin the circle from to top/center like in the design?
![image](https://user-images.githubusercontent.com/1374174/41594473-73f9368e-73bb-11e8-85b5-727342e46f49.png)